fix(BLink): prefer passed active prop over isActive slot prop from RouterLink

tvaliasek opened the pull request:
# Describe the PR

BLink component ignores directly set active prop, when router link passes isActive prop, this causes invalid assignment of active class when nested routes is used.

## Small replication

See tests, or:

Create vue instance with vue-router and nested routes and two BLinks, one heading to default route target and with active prop set to false, and then second to its nested route. After navigation to second route, both BLinks has active class assigned, but the first BLink shouldn't have.

## PR checklist

<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

VividLemon commented:
https://router.vuejs.org/guide/migration/#removal-of-the-exact-prop-in-router-link

Remove this prop. It should not be doing anything

VividLemon commented on packages/bootstrap-vue-next/src/components/BLink/BLink.vue:
```diff
@@ -70,6 +70,24 @@ export default defineComponent({
     const link = ref<HTMLElement>(null as unknown as HTMLElement)
 
+    const isActivePassed = computed(() => {
+      if (props.active === undefined) {
```
`useBooleanish` has overloads that allow for passing undefined directly. Where the output would be boolean | undefined...

With this in mind, you likely don't need to do all this manipulation, you could instead just do 
```html
(activeBoolean ?? isActive) && activeClass,
```
I believe this would have the same affect, in a more concise syntax.

tvaliasek replied:
Fixed as suggested. The tests passed without errors, so it looks fine.

tvaliasek commented:
> https://router.vuejs.org/guide/migration/#removal-of-the-exact-prop-in-router-link
> 
> Remove this prop. It should not be doing anything

Removed exact prop, also removed from dependent BButton component.

VividLemon commented on packages/bootstrap-vue-next/src/components/BLink/BLink.vue:
```diff
@@ -39,12 +36,11 @@ import {computed, defineComponent, getCurrentInstance, type PropType, ref, toRef
   disabled: {type: [Boolean, String] as PropType<Booleanish>, default: false},
   event: {type: [String, Array], default: 'click'},
-  exact: {type: [Boolean, String] as PropType<Booleanish>, default: false},
   exactActiveClass: {type: String, default: 'router-link-exact-active'},
```
Subsequently exactActiveClass should be removed. It serves no real purpose that couldn't be resolved from using $attrs

tvaliasek replied:
Ok, removed from both BLink and BButton.
